mmeval/metrics/instance_seg.py

```diff
@@ -0,0 +1,288 @@
+                                 and valid_class_ids is not None)
+        if self.dataset_meta is not None:
+            if classes is not None or valid_class_ids is not None:
+                warnings.warn('`classes` and `valid_class_ids` are ignored.')
+            self.classes = self.dataset_meta['classes']
+            self.valid_class_ids = self.dataset_meta['seg_valid_class_ids']
+        else:
+            if not isinstance(classes, list):
+                raise TypeError('`classes` should be List[str]')
+            if not isinstance(valid_class_ids, list):
+                raise TypeError('`valid_class_ids` should be List[int]')
+            self.classes = classes
+            self.valid_class_ids = valid_class_ids
+
+    def add(self, predictions: Sequence[Dict], groundtruths: Sequence[Dict]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Process one batch of data samples and predictions.
+
+        The processed results should be stored in ``self.results``,
+        which will be used to compute the metrics when all batches
+        have been processed.
+
+        Args:
+            predictions (Sequence[Dict]): A sequence of dict. Each dict
+                representing a detection result, with the following keys:
+
+                - pts_instance_mask(np.ndarray): Predicted instance masks.
+                - instance_labels(np.ndarray): Predicted instance labels.
+                - instance_scores(np.ndarray): Predicted instance scores.
+
+            groundtruths (Sequence[Dict]): A sequence of dict. Each dict
+                represents a groundtruths for an image, with the following
+                keys:
+
+                - pts_instance_mask(np.ndarray): Ground truth instance masks.
+                - pts_semantic_mask(np.ndarray): Ground truth semantic masks.
+        """
+        for prediction, groundtruth in zip(predictions, groundtruths):
+            self._results.append((deepcopy(prediction), deepcopy(groundtruth)))
+
+    def compute_metric(self, results: List[List[Dict]]) -> Dict[str, float]:
+        """Compute the metrics from processed results.
+
+        Args:
+            results (list): The processed results of each batch.
+
+        Returns:
+            Dict[str, float]: The computed metrics. The keys are the names of
+            the metrics, and the values are corresponding results.
+        """
+        gt_semantic_masks = []
+        gt_instance_masks = []
+        pred_instance_masks = []
+        pred_instance_labels = []
+        pred_instance_scores = []
+
+        for result_pred, result_gt in results:
+            gt_semantic_masks.append(result_gt['pts_semantic_mask'])
+            gt_instance_masks.append(result_gt['pts_instance_mask'])
+            pred_instance_masks.append(result_pred['pts_instance_mask'])
+            pred_instance_labels.append(result_pred['instance_labels'])
+            pred_instance_scores.append(result_pred['instance_scores'])
```

Comment:
A suggestion that simplify the code~
```suggestion
        gt_semantic_masks = [gt['pts_semantic_mask'] for _, gt in results]
        gt_instance_masks = [gt['pts_instance_mask'] for _, gt in results]
        pred_instance_masks = [pred['pts_instance_mask'] for pred, _ in results]
        pred_instance_labels = [pred['instance_labels'] for pred, _ in results]
        pred_instance_scores = [pred['instance_scores'] for pred, _ in results]
```

Reply:
In this way, we would do 5 for loops, which may be time consuming.